department-of-veterans-affairs/caseflow
# Add ‘Issue Type’ condition for report

Resolves: https://jira.devops.va.gov/browse/APPEALS-30939
Main Jira effort: https://jira.devops.va.gov/browse/APPEALS-26499

# Description
Add the `Issue Type` condition field input for the Generate Task Report effort.

## Acceptance Criteria
- [ ] If user selects ‘Issue Type’ condition, issue types selection drop-down appears.
- [ ] Issue Type data will be populated using new JSON/constants file created in [APPEALS-30949](https://jira.devops.va.gov/browse/APPEALS-30949)
- [ ] User can type to find Issue Type name or select from drop-down.
- [ ] Drop-down options for ‘Issue Type’ include: list of VHA approved facilities (see below))
- [ ] User can type to search or select more options from the drop-down selection to include all desired options.
- [ ] User can remove any selected Issue Types displayed via the 'X' next to each selection. 
- [ ] At least one Issue Type must be selected if condition is added

## Testing Plan
- Log in as VHAADMIN1
- Go to decision_reviews/vha/report
- Click Add Condition
- Select Issue Type
- See the dropdown and interact with it
- Click submit and check the dev tools console to see if the correct data was logged

### Screenshots
![Screenshot 2023-12-01 at 3 59 56 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/1078402/24d645d0-e0c8-40f0-8c57-8b571aac316c)
